src/lib/API/api.js

```diff
@@ -531,7 +539,10 @@ export class APIService {
       })
 
       params.page += 1
-      txs.push(...events)
+
+      if (isArray(events)) {
```

Comment:
still not correct.
if not array need special handling. see next lines that do events.length assuming it is an array.
also it means there's an error, maybe rate limit? does the code that uses it will retry? relies on running in intervals?
all of that needs to be verified and checked that our logic stays correct.

did you verify the lastBlock check I mentioned?

Reply:
1. all api explorer calls now have a array check
2. I believe to understand in the flow now it is handling a 'retry' based on the final line of syncTxFromExplorer:
```
    const lastBlock = Number(last(results)?.blockNumber)
    return lastBlock ? lastBlock + 1 : startBlock
```
since it resets to the original start-block when no blockNumber can be determined so will retry based on the polling interval set

3. see comment: